core/src/Cabana_Parallel.hpp

```diff
@@ -202,6 +210,67 @@ inline void neighbor_parallel_for(
         } );
 }
 
+//---------------------------------------------------------------------------//
+/*!
+  \brief Execute \c functor in parallel according to the execution \c policy
+  with a thread-local serial loop over particle neighbors and serial loop over
+  particle angular neighbors.
+
+  \tparam ExecutionSpace The execution space in which to execute the functor.
+
+  \tparam FunctorType The functor type to execute.
+
+  \tparam NeighborListType The neighbor list type.
+
+  \param exec_policy The policy over which to execute the functor.
+
+  \param functor The functor to execute in parallel
+
+  \param list The neighbor list over which to execute the neighbor operations.
+
+  \param tag Algorithm tag indicating a serial loop strategy over particle
+  neighbors.
+
+  \param tag Algorithm tag indicating a serial loop strategy over particle
+  angular neighbors.
+
+  \param str An optional name for the functor. Will be forwarded to the
+  Kokkos::parallel_for called by this code and can be used for identification
+  and profiling purposes.
+*/
+template <class FunctorType, class NeighborListType, class... ExecParameters>
+inline void neighbor_parallel_for(
+    const Kokkos::RangePolicy<ExecParameters...> &exec_policy,
+    const FunctorType &functor, const NeighborListType &list,
+    const SerialNeighborOpTag, const SerialAngularNeighborOpTag,
```

Comment:
Yeah I think we should change tag names as I described - otherwise this is rather confusing. Then we have one tag specifying parallelism and another specifying the neighbors we loop over.